### PR TITLE
Calendar Storage 

### LINKED ISSUE #41: Save and Load Tasks
Need to discuss how to do

### LINKED ISSUE #42: Task Serialization
TaskSerialiser will encode and decode Task objects into string objects. #44

### PR DESCRIPTION
**Pull request by 9gix**
This will closes #42 closes #41 
- [x] saving of calendar
- [x] loading of calendar
- [x] load calendar when the application start
- [x] save the calendar every time command executed

### CODE REVIEW

**Comment by starkipraggy** [comment]
Why is there a pull request for this when it's not done yet?

**Comment by 9gix** [comment]
@starkipraggy We do not use PR solely for for merging, but also for tracking the progress (with checklist) and reviewing (with diff changes of the branch)

**Comment by starkipraggy** [comment]
The code looks like it checks out. I can't run now. Will update again.

**Comment by bsmmoon** [comment]
I did copy vendor-libs to Library but there is no difference. Still has 387 Errors.

**Comment by bsmmoon** [comment]
oh wait.... boost.............................

**Comment by 9gix** [comment]
If anyone do encounter any errors, please elaborate some of the key errors. So I could improve the docs for the setup

File: DoLah/Models/ISerializable.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+#include <string>
+
+class ISerializable {
+public:
+    virtual ISerializable* serialize(const std::string&) = 0;
+    virtual std::string deserialize() = 0;
```
Comment (bsmmoon) [inline DoLah/Models/ISerializable.h]: here!

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[submodule "yaml-cpp"]
```
Comment (9gix) [inline .gitmodules]: @bsmmoon what is this

**Comment by 9gix** [comment]
@eatedcookie https://github.com/cs2103aug2015-w11-2c/DoLah/commit/79e2fa2a0ffabaab241c89a781e0737a087720fb

**Pull request merged**